watertap/core/zero_order_sido_reactive.py

```diff
@@ -0,0 +1,416 @@
+    except KeyError:
+        raise KeyError(
+            f"{self.name} - database provided does not contain a list of "
+            f"reactions for this technology.")
+    # Create indexing set for reactions
+    self.reaction_set = Set(initialize=rxn_ids)
+
+    # Add performance variables
+    self.recovery_frac_mass_H2O = Var(
+        self.flowsheet().time,
+        initialize=0.8,
+        domain=NonNegativeReals,
+        units=pyunits.dimensionless,
+        bounds=(1E-8, 1.0000001),
+        doc='Mass recovery fraction of water in the treated stream')
+    self.removal_frac_mass_solute = Var(
+        self.flowsheet().time,
+        self.config.property_package.solute_set,
+        domain=NonNegativeReals,
+        initialize=0.01,
+        units=pyunits.dimensionless,
+        doc='Solute removal fraction on a mass basis')
+
+    self.reaction_conversion = Var(
+        self.flowsheet().time,
+        self.reaction_set,
+        initialize=0.5,
+        units=pyunits.dimensionless,
+        doc='Reaction conversion based on key reactant')
+
+    # Load reaction conversions
+    for (t, r), v in self.reaction_conversion.items():
+        try:
+            conv = dbparams["reactions"][r]["conversion"]
+        except KeyError:
+            raise KeyError(
+                f"{self.name} - database provided does not "
+                f"contain an entry for conversion for reaction {r}.")
+        v.fix(conv)
+
+    # Intermediate variables and parameters
+    self.extent_of_reaction = Var(
+        self.flowsheet().time,
+        self.reaction_set,
+        initialize=0,
+        units=pyunits.kg/pyunits.s,
+        doc='Reaction extent on a mass basis based on key reactant')
+
+    # Reaction extent equation
+    @self.Constraint(self.flowsheet().time,
+                     self.reaction_set,
+                     doc='Calcuation of reaction extent from conversion')
+    def reaction_extent_equation(b, t, r):
+        # Get key reactant from database
+        try:
+            key_reactant = dbparams["reactions"][r]["key_reactant"]
+        except KeyError:
+            raise KeyError(
+                f"{self.name} - database provided does not "
+                f"contain an entry for key_reactant for reaction {r}.")
+        if key_reactant not in b.properties_in[0].component_list:
+            raise ValueError(
+                f"{self.name} - key_reactant {key_reactant} for reaction {r} "
+                f"is not in the component list used by the assigned property "
+                f"package.")
+        return (b.extent_of_reaction[t, r] ==
+                b.reaction_conversion[t, r] *
+                b.properties_in[t].flow_mass_comp[key_reactant])
+
+    self.generation_ratio = Param(
+        self.reaction_set,
+        self.config.property_package.component_list,
+        initialize=0,
+        mutable=True,
+        units=pyunits.dimensionless,
+        doc="Mass ratio for generation of species w.r.t. key species")
```

Comment:
Is there a reason you chose `generation_ratio` rather than `conversion_ratio`? The temporary variable you use to calculate this is `cratio` and the comments refer to it as the conversion ratio.

Reply:
Why did you decide to go with a param here? Isn't the IDAES convention to just use fix vars?

Reply:
1. I think the two terms are essentially synonymous, and that is just what I picked.
2. No, the IDAES convention is to use Vars for anything that might conceivably be solved for, however this is something that definitely should not be left free. Hence the use of a Param.